web/src/components/tasks/anytime-action/tax-clearance-certificate/AnytimeActionTaxClearanceCertificateElement.tsx

```diff
@@ -175,15 +175,18 @@ export const AnytimeActionTaxClearanceCertificateElement = (props: Props): React
                   <Heading level={1}>{props.anytimeAction.name}</Heading>
                 </div>
               </div>
-              {certificatePdfArray ? (
-                <TaxClearanceDownload />
+              {certificatePdfBlob ? (
+                <TaxClearanceDownload
+                  certificatePdfBlob={certificatePdfBlob}
+                  downloadFilename={`Tax Clearance Certificate - ${Date.now()}`}
+                />
               ) : (
                 <TaxClearanceSteps
                   steps={stateTaxClearanceCertificateSteps}
                   currentStep={stepIndex}
                   stepIndex={setStepIndex}
                   saveTaxClearanceCertificateData={saveTaxClearanceCertificateData}
-                  certificatePdfArray={setCertificatePdfArray}
+                  setCertificatePdfBlob={setCertificatePdfBlob}
```

Comment:
[dust] This is a lot of information to be sending through props, which may impact performance. Is this a case where `useContext` could lead to better performance within this component and its children?